services/media/service/src/common/utils/token-utils.ts

```diff
@@ -5,8 +5,13 @@ import {
 import NodeCache from 'node-cache';
 import { Config } from '../config';
 
-// By default, service account token is valid for 1 hour
-const cache = new NodeCache({ stdTTL: 55 * 60 });
+const tokenExpirationInSeconds = 7 * 24 * 60 * 60; // 7 days
```

Comment:
If we want to cache it by ingest document - we would have to pass ingest document id as part of `ingest_correlation_id` property, in addition to the current value of `Ingest Item ID`. Because the place where we call the getServiceToken function is the message handler after the db trigger. Do we want to go this way?

Reply:
For now, token is valid for 7 days and refreshes every hour. Expecting a more sophisticated cache invalidation approach in the future.